### DISCUSSION TITLE
Skypilot agent

### DISCUSSION THREAD

**Discussion by aswanthkrishna** [flyteorg/flyte#4645] [Ideas]
i came accross this pretty useful project [skypilot](https://github.com/skypilot-org/skypilot). was wondering if it makes sense to have some sort of integration with flyte.

**Comment by kumare3** [comment]
If you would be willing to then a Flyte agent would be a cool integration

**Reply by aswanthkrishna** [reply]
yeah would love to have a go at it.

**Reply by kumare3** [reply]
Talk to @samhita-alla and use the memverge agent as an example

**Reply by samhita-alla** [reply]
here's the memverge agent code: https://github.com/flyteorg/flytekit/tree/master/plugins/flytekit-mmcloud

**Reply by aswanthkrishna** [reply]
@samhita-alla i got around writing the plugin, can you point me towards any documentation for deploying it in sandbox cluster? i tried to follow [this](https://www.youtube.com/watch?v=nD98GQ-pyAE&t=848s) but wasnt successful.

**Reply by samhita-alla** [reply]
@aswanthkrishna have you looked at https://docs.flyte.org/en/latest/flytesnacks/examples/development_lifecycle/agent_service.html#extend-agent-service doc? there's also https://hackmd.io/@flyte-community/ryaSdNtmT doc that should help you test the agent locally.

**Reply by pingsutw** [reply]
@aswanthkrishna Feel free to DM me on Slack (Kevin Su), I can help you if you encounter any issues.

**Reply by Future-Outlier** [reply]
@aswanthkrishna 
- These 2 docs can help you.
https://docs.flyte.org/en/latest/flytesnacks/examples/development_lifecycle/agent_service.html#extend-agent-service
https://hackmd.io/X4ah4tTUQXKiDf5RA1BSIA
- These 2 PRs can help you.
sync task: https://github.com/flyteorg/flytekit/pull/2086
async task: https://github.com/flyteorg/flytekit/tree/master/plugins/flytekit-bigquery
- These 2 talks can help you.
https://youtu.be/nD98GQ-pyAE?si=-epRr3uKH9p14oh2
https://youtu.be/IS6gi4jR7h0?si=OodX9IRThNlxjqkc

**Reply by aswanthkrishna** [reply]
thanks @pingsutw and @Future-Outlier for the resources.
